test/unit/search/search_scheme_algorithm_test.cpp

```diff
@@ -0,0 +1,329 @@
+        auto const & hits_tmp = it.locate();
+        hits_ss.insert(hits_ss.end(), hits_tmp.begin(), hits_tmp.end());
+    };
+
+    uint8_t const total        = search.u.back();
+    uint8_t const substitution = std::rand() % (total + 1);
+
+    detail::search_param error_left{total, substitution, 0, 0};
+
+    // Find all hits using search schemes.
+    detail::search_ss<false>(it, query, start_pos, start_pos + 1, 0, 0, true, search, blocks_length, error_left,
+                             delegate_ss);
+    // Find all hits using trivial backtracking.
+    detail::search_trivial<false>(it, query, 0, error_left, delegate_trivial);
+
+    // Eliminate hits that we are not interested in (based on the search and chosen error distribution)
+    hits_ss.erase(std::remove_if(hits_ss.begin(), hits_ss.end(),
+                                 [&text, &orig_query, query_length](uint64_t const hit)
+                                 {
+                                     dna4_vector matched_seq = text | ranges::view::slice(hit, hit + query_length);
+                                     return (matched_seq != orig_query);
+                                 }), hits_ss.end());
+    hits_trivial.erase(std::remove_if(hits_trivial.begin(), hits_trivial.end(),
+                                      [&](uint64_t const hit)
+                                      {
+                                          // filter only correct error distributions
+                                          dna4_vector matched_seq = text | ranges::view::slice(hit, hit + query_length);
+                                          if (orig_query != matched_seq)
+                                              return true;
+
+                                          uint64_t lb = 0, rb = 0;
+                                          uint8_t total_errors = 0;
+                                          for (uint8_t block = 0; block < search.blocks(); ++block)
+                                          {
+                                              rb += ordered_blocks_length[block];
+
+                                              uint8_t errors = 0;
+                                              for (uint64_t i = lb; i < rb; ++i)
+                                                  if (hit + i >= text.size())
+                                                      ++errors;
+                                                  else
+                                                      errors += query[i] != text[hit + i];
+                                              total_errors += errors;
+                                              if (errors != error_distribution[block])
+                                                  return true;
+                                              lb += ordered_blocks_length[block];
+                                          }
+                                          return false;
+                                      }), hits_trivial.end());
```

Comment:
you put the two-line lambas outside, but define this 25-line lambda inline? :woman_facepalming: